Improve the Features section of the documentation

settermjd opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

The intention of this change is to improve the readability, formatting, and pitch of the [Features](https://docs.mezzio.dev/mezzio/v3/getting-started/features/) section of the Getting Started section of the documentation.

It comprises a small set of changes for a series of different reasons.

Firstly, as this is likely one of the first sections of the docs a new/first-time reader may go to, the language has been reworked to be a bit more sales-like and a little less technical. This is because this section of the docs should really tell the reader what's in it for them, encouraging them to keep exploring and learning.

Secondly, it changes the section's headers from bullet points to level three headers. This makes them both easier to read and easier for the reader to scan.

Thirdly, from the perspective of maintenance, sentences have been changed to be all one one line, not split up over multiple lines and the links have been extracted from being directly embedded with the text to be [reference-style links](https://www.markdownguide.org/basic-syntax/#reference-style-links). Personally, I find these easier to work with and maintain.

settermjd committed:
Change feature "headers" to level 3 headers

As it was, as bolded bullet points for each of the four key features,
didn't look well-formatted, nor all that professional. This change
converts the names of each of the key four features to level three
headers to correct that, in the process also improve accessibility.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd committed:
Update the Features section of the Getting Started guide

This change comprises a small set of changes for a series of different
reasons.

Firstly, as this is likely one of the first sections of the docs a
new/first-time reader may go to, the language has been reworked to be a
bit more sales-like and a little less technical. This is because this
section of the docs should really tell the reader what's in it for them,
encouraging them to keep exploring and learning.

Secondly, it changes the section's headers from bullet points to level
three headers. This makes them both easier to read and easier for the
reader to scan.

Thirdly, from the perspective of maintenance, sentences have been
changed to be all one one line, not split up over multiple lines and the
links have been extracted from being directly embedded with the text to
be reference-style links
(https://www.markdownguide.org/basic-syntax/#reference-style-links).
Personally, I find these easier to work with and maintain.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd commented:
Unless I've missed something, the QA checks that fail have nothing to do with the changes I made.

gsteel commented:
If you merge my lint fixing pr and rebase this it should go green 👍

gsteel commented on docs/book/v4/getting-started/features.md:
```diff
@@ -30,45 +30,32 @@ With Mezzio, you can build middleware applications such as the following:
-  routing library that best fits the project needs. By default, we provide
-  wrappers for Aura.Router, FastRoute, and the laminas-router.
+Routing is abstracted in Mezzio, allowing the developer to choose the routing library that best fits the project needs.
+By default, we provide wrappers for [Aura.Router][aura-router-url], [FastRoute][fastroute-url], and the [laminas-router][laminas-router-url].
```
Aura Router is dead: https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2024-08-12-TSC-Minutes.md#abandon-mezziomezzio-aurarouter

settermjd replied:
Thanks for that. My original intention was only to refresh what was already there. But, taking out deprecated packages makes total sense.

gsteel commented on docs/book/v4/getting-started/features.md:
```diff
@@ -30,45 +30,32 @@ With Mezzio, you can build middleware applications such as the following:
-  specialized error response generators that can perform templating or use
-  Whoops.
+Applications should handle errors gracefully, but also handle them differently in development versus production.
+Mezzio provides both basic error handling via Stratigility's own `ErrorHandler` implementation, providing specialized error response generators that can perform templating or use [Whoops][whoops-url].
```
Might be worth mentioning the problem-details lib here which solves error handling for apis in a consistent way

settermjd replied:
I'll come back to this in a future PR, if that's okay.

settermjd committed:
Update the Features section of the Getting Started docs

This change is intended to make the section as clear and engaging as
possible, while still retaining the core technical details.

settermjd commented:
> If you merge my lint fixing pr and rebase this it should go green 👍

I, perhaps mistakenly, targeted my PR at 4.0.x, and your PR's merged into 3.2.x. Should I retarget this PR at that branch instead?

gsteel reviewed (approved):
LGTM! Thanks @settermjd

gsteel commented:
> I, perhaps mistakenly, targeted my PR at 4.0.x, and your PR's merged into 3.2.x. Should I retarget this PR at that branch instead?

Ah yes! Not sure - The problem is that if you switch to 3.x you'll have to reapply the changes to the docs tree in `./docs/v3` - perhaps better to merge as-is into 4.x and deal with the unrelated linting issues separately. Obviously, the improvements you've made won't be visible until 4.0 is realeased.

cc @froschdesign

settermjd commented:
> > I, perhaps mistakenly, targeted my PR at 4.0.x, and your PR's merged into 3.2.x. Should I retarget this PR at that branch instead?
> 
> Ah yes! Not sure - The problem is that if you switch to 3.x you'll have to reapply the changes to the docs tree in `./docs/v3` - perhaps better to merge as-is into 4.x and deal with the unrelated linting issues separately. Obviously, the improvements you've made won't be visible until 4.0 is realeased.
> 
> cc @froschdesign

Yeah, my mistake. I'll apply the same changes, including yours, to the 3.20.x branch in a new PR.
